service/dps/storm/common/src/main/java/eu/europeana/cloud/service/dps/storm/utils/TaskStatusChecker.java

```diff
@@ -1,34 +1,41 @@
 package eu.europeana.cloud.service.dps.storm.utils;
 
-import com.google.common.cache.Cache;
 import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
 import eu.europeana.cloud.cassandra.CassandraConnectionProvider;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.concurrent.ConcurrentMap;
-
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Created by Tarek on 4/9/2018.
  */
 public class TaskStatusChecker {
-    public static final int CHECKING_INTERVAL = 5000;
+
+    public static final int CHECKING_INTERVAL_IN_SECONDS = 5;
     public static final int CONCURRENCY_LEVEL = 1000;
     public static final int SIZE = 100;
 
     private static TaskStatusChecker instance;
     private CassandraTaskInfoDAO taskDAO;
 
 
-    private static volatile Cache<Long, CacheItem> cache;
+    private static volatile LoadingCache<Long, Boolean> cache;
+
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TaskStatusChecker.class);
 
     private TaskStatusChecker(CassandraConnectionProvider cassandraConnectionProvider) {
 
-        cache = CacheBuilder.newBuilder().concurrencyLevel(CONCURRENCY_LEVEL).maximumSize(SIZE).softValues()
-                .build();
+        cache = CacheBuilder.newBuilder().refreshAfterWrite(CHECKING_INTERVAL_IN_SECONDS, TimeUnit.SECONDS).concurrencyLevel(CONCURRENCY_LEVEL).maximumSize(SIZE).softValues()
```

Comment:
Static field, initialize statically?
This probably causes other issues but what about the following:
Perhaps creating an inner class and include the isTaskKilled inside that class while making taskDao field static and initializing taskDaon in the init method, and also removing the parameter from the constructor?

Does that make sense, what do you think?

Reply:
Not realy, This is a Singlton class, and this constructor will be executed only once per JVM. 
we need a constructor to create a utility class and to insure the passing of CassandraConnectionProvider(which could change from one JVM to another)

Reply:
Yes it's Singleton but the CassandraConnectionProvider is used for the taskDao that could be initialized on the init method just after calling the new TaskStatusChecker()(without parameter) . Anyway, I tried it in code and seems fine, but up to you to decide.